### ISSUE TITLE
On wit sintax: formalizing documentation comments?

### ISSUE THREAD

**Issue by saona-raimundo** [bytecodealliance/wit-bindgen#117]
Hi there!

Thank you for the good work!

I read #108 and @sunfishcode mentioned there is interest in feedback on the current [wit sintax](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md).

I wanted to ask about [comment in wit sintax](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md#comments):
Should there be documentation and code comments?

## Current state

While trying out `wit-bindgen`, I noticed that the code generation for Markdown passes comments as documentation, which is really nice! The online demo accepts both `//` and `///` for (documentation) comments when generating the `.md` file.

I really like the current behaviour: taking comments before an item as its documentation. 

As an example of what I would write and use the Markdown generated file as documentation:
``` wit
/// person represents a physical, real person
record person {
  /// Legal name of the person
  name: string,
  /// Year of bird
  year: u32,
  // TODO: add more fields as they are needed
}

// This comment does not appear in the md file generated because there is no element that follows

/// Returns the greeting we shoud use with `person`.
hello: function(who: person) -> string
```

## "Use case"

For now, `.wit` files are expected to be small(?) so this difference is not really needed. But if they get bigger and more complex, this differentiation of comments might come in handy :)

Currently, "code comments" can be achieved by adding a new line after them: no element follows this comment, so they are not passed to the `.md` file. But maybe this behaviour of the `wit-bindgen-gen-markdown` should be included in the wit syntax?

**Comment by alexcrichton** [comment]
As you've noticed the `*.wit` format supports comments of the form `//` and pseudo-supports `///` for "doc comments". In that sense can you describe a bit more what else you'd like to see?

**Comment by saona-raimundo** [comment]
Thank you for the answer! I will try to keep it short this time then.

Shall the [`WIT.md` document](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md) formally include "doc-comments" of the form `///`?

I am not asking for any functionality, just asking if this formalization is useful at this point.

**Comment by alexcrichton** [comment]
Seems reasonable to me!

**Comment by saona-raimundo** [comment]
To implement this, one would need to skip implementation comments (`//` and `/* ... */`) in markdown and only consider documentation comments (`//`).

I am not sure how to do it myself, but I can try if a PR for this would be welcome :)

**Comment by alexcrichton** [comment]
Sure yeah, a PR would be great!

**Comment by Kylebrown9** [comment]
Was this resolved by #128?

**Comment by alexcrichton** [comment]
Indeed!
